packages/discord.js/src/structures/CategoryChannel.js

```diff
@@ -1,19 +1,20 @@
 'use strict';
 
 const GuildChannel = require('./GuildChannel');
+const CategoryChannelChildManager = require('../managers/CategoryChannelChildManager');
 
 /**
  * Represents a guild category channel on Discord.
  * @extends {GuildChannel}
  */
 class CategoryChannel extends GuildChannel {
   /**
-   * Channels that are a part of this category
-   * @type {Collection<Snowflake, GuildChannel>}
+   * A manager of the channels belonging to this category
+   * @type {CategoryChannelChildManager}
    * @readonly
    */
   get children() {
-    return this.guild.channels.cache.filter(c => c.parentId === this.id);
+    return new CategoryChannelChildManager(this);
   }
```

Comment:
This doesn't need to be a getter anymore

Reply:
I was told to make it a getter by @kyranet though ?

Reply:
The problem with making it a property is that it makes the memory usage higher, @vladfrangu. This is a perfectly computed class, it can be a getter just fine.

Reply:
How does it make memory usage higher? Does it even matter? Creating an instance of this class every time you access it doesn't seem like a smart idea either, and it doesn't store any data itself 🤷

Reply:
Well, this class stores `guild` and `channel`.

Then inherits `CachedManager`, which creates a Collection (`_cache`) no matter what, with `Object.defineProperty`:

https://github.com/discordjs/discord.js/blob/cd2f5660521ae4a668727cd246c6e5b0abe53219/packages/discord.js/src/managers/CachedManager.js#L14

Then inherits `DataManager` (which this class should extend anyways), and this stores `holds` using `Object.defineProperty`

https://github.com/discordjs/discord.js/blob/cd2f5660521ae4a668727cd246c6e5b0abe53219/packages/discord.js/src/managers/DataManager.js#L22

And it finally extends `BaseManager`, which creates a property of `client`, also with `Object.defineProperty`:

https://github.com/discordjs/discord.js/blob/cd2f5660521ae4a668727cd246c6e5b0abe53219/packages/discord.js/src/managers/BaseManager.js#L15

Now that's a total of 5 properties, one of them being a brand new `Map` (which are very heavy), and 3 of those properties are done so with `Object.defineProperty` (and you know the performance implications of this).

Making it a property will force the code to always run, resulting on a large performance and memory impact as categories are on average the 17% of all channels (at least on my bot, out of 384,183 channels, 65,587 are category channels).

Furthermore, `CategoryChannel#children` isn't that used, nor it's that useful, it's a utility that provides nothing but convenience. *As a matter of a fact, I haven't even used this property in my [bot's code](https://github.com/skyra-project/skyra/search?l=TypeScript&q=.children), which proves that even kitchensinks may not use it.* I have yet to see real world usage that uses this property efficiently.

If we are to lower the CPU and RAM usage of discord.js, we should start by not forcing the storage of data that not everyone needs.